internal/git/config.go

```diff
@@ -9,16 +9,33 @@ import (
 	"github.com/pkg/errors"
 )
 
+//go:generate moq -out config_mock.go . GitConfigAPI
+
+// GitConfigAPI is an interface to interact with a git config system
+// this makes it possible to extract information from the repository
+// or the local user
+type GitConfigAPI interface {
+	CommitterDetails() (name string, email string, err error)
+}
+
+type LocalGitConfigAPI struct {
+}
+
+// NewLocalGitConfigAPI provide a GitConfigAPI for a local repository
+func NewLocalGitConfigAPI() GitConfigAPI {
+	return &LocalGitConfigAPI{}
+}
+
 // CommitterDetails returns name and email read for a Git configuration file.
 //
 // Fetching the configuration values are delegated to the git CLI and follows
 // precedence rules defined by Git.
-func CommitterDetails() (string, string, error) {
-	name, err := getValue("user.name", "HAMCTL_USER_NAME")
+func (*LocalGitConfigAPI) CommitterDetails() (name string, email string, err error) {
```

Comment:
Any reason for using named return types? It always sticks out to me 🙈

Reply:
Yeah, I am not really using the name types pr. say, I just use it to document the 'types'. I find it confusing that you can have multiple strings, but they semantically have different domain models. I.e. one is a name, another an email and so on.

Reply:
Makes sense. I guess what would be "nice" is to actually introduce a struct to name the fields? I will go a long way to avoid named return values 🙈

Reply:
It has been done =D